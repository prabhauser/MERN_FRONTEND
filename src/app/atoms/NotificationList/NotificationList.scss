@import '../../../utilities/theme/_theme.scss';
@import '../../../assets/fonts/typography.scss';

.date {
    color: #5c697c;
    @extend .c-poppins-regular;
    font-size: 10px;
    font-weight: 500;
}
.dateCont {
    display: flex;
    justify-content: space-between;
}
.description {
    color: $s200Bg;
    @extend .c-poppins-regular;
    font-size: 10px;
    font-weight: 500;
}
.title {
    color: $s900Bg;
    @extend .c-poppins-regular;
    font-size: 12px;
    font-weight: 600;
    margin: unset;
}
.notifContiner {
    display: flex;
}
.body {
    height: 360px;
    width: 333px;
    display: contents;
}
