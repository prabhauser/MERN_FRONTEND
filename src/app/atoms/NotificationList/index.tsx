import React from 'react';
import MenuItem from '@mui/material/MenuItem';
import Box from '@mui/material/Box';
import ColorBellIcon from '../../../assets/images/ColorBell.svg';
import './NotificationList.scss';
type Props = {
    item: any;
    handleCloseUserMenu: any;
};

const NotificationList = (props: Props) => {
    const { item, handleCloseUserMenu } = props;
    return (
        <MenuItem
            key={item.id}
            onClick={handleCloseUserMenu}
            style={{ width: '250px', borderBottom: '1px solid rgba(204,208,213,0.6)' }}
        >
            <Box className="notifContiner">
                <Box style={{ width: '50px', display: 'flex', alignItems: 'center' }}>
                    <img src={ColorBellIcon} alt="bell" />
                </Box>
                <Box style={{ width: '168px' }}>
                    <h5 className="title">{item.Title}</h5>
                    <span className="description">{item.Content}</span>
                    <div className="dateCont">
                        <span className="date">{item.Date}</span>
                        <span className="date">time</span>
                    </div>
                </Box>
            </Box>
        </MenuItem>
    );
};

export default NotificationList;
