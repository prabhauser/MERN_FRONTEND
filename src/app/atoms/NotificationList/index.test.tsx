import { shallow } from 'enzyme';
import NotificationList from './index';

describe('<NotificationList />', () => {
    it('should take NotificationList snapshot', () => {
        const component = shallow(<NotificationList item={'Notificationvalues'} handleCloseUserMenu={() => {}} />);
        component.update();
        expect(component).toMatchSnapshot();
    });
});
