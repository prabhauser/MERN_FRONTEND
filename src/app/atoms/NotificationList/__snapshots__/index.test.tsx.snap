// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationList /> should take NotificationList snapshot 1`] = `
<ForwardRef(MenuItem)
  onClick={[Function]}
  style={
    Object {
      "borderBottom": "1px solid rgba(204,208,213,0.6)",
      "width": "250px",
    }
  }
>
  <ForwardRef(Box)
    className="notifContiner"
  >
    <ForwardRef(Box)
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "50px",
        }
      }
    >
      <img
        alt="bell"
        src="ColorBell.svg"
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)
      style={
        Object {
          "width": "168px",
        }
      }
    >
      <h5
        className="title"
      />
      <span
        className="description"
      />
      <div
        className="dateCont"
      >
        <span
          className="date"
        />
        <span
          className="date"
        >
          time
        </span>
      </div>
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(MenuItem)>
`;
